### PR TITLE
chore(test): add test infra mysql run cleanup script

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Adds script to periodically clean up mysql run records to prevent spurious sample test failures.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about 
the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: test/tools/project-cleaner/delete_old_mysql_records.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if not next_page_token:
+            break
+    return ids
+
+
+def delete_old_runs(kfp_client: client.Client, after: datetime.datetime) -> int:
+    ids_to_delete = accumulate_ids_to_delete(kfp_client=kfp_client, after=after)
+    deleted_run_count = 0
+    for run_id in ids_to_delete:
+        kfp_client.delete_run(run_id)
```
Comment (zijianjoy) [inline test/tools/project-cleaner/delete_old_mysql_records.py:82]: How does KFP client authenticate user access to KFP instance?

File: test/tools/project-cleaner/delete_old_mysql_records.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        deleted_run_count += 1
+    return deleted_run_count
```
Comment (zijianjoy) [inline test/tools/project-cleaner/delete_old_mysql_records.py:84]: Do we need to count deleted run here? Can we just use `return len(ids_to_delete)`

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Comment by connor-mccarthy** [comment]
Closing, preferring manual run of the cleanup script in the short term.

**Pull request merged**